File: cpp/linkedlist.cpp
#include<iostream>

using namespace std;

struct linkedlist{
	int data;
	struct linkedlist *next;

};
int main(){
	linkedlist head;
	struct linkedist *p;
	p = new linkedlist;
	head.data = 6;
	linkedlist node;
	node.data = 78;
	head.p = &node;
	linkedlist *curr;
	curr = &head;

	while(curr != NULL){
		cout << curr->data;
		curr = curr->p;
		cout << "->";
	}

}

File: practice/LinkedList/LinkedList.java
class LinkedList{
	class Node {
		int data;
		Node next = null;

		Node(int data){
			this.data = data;
		}
	}

	Node head;
	LinkedList(int data){
		this.head = new Node(data);
	}

	public void add(int data){
		Node node = new Node(data);
		if(head == null){
			head = node;
			return;
		}
		Node curr = head;

		while(curr.next != null){
			curr = curr.next;
		}

		curr.next = node;

	}

	public void pop(){
		if(head == null){
			System.out.println("List empty");
			return;
		}
		if(head.next == null){
			head = null;
			return;
		}

		Node curr = head;
		Node prev = null;
		while(curr.next != null){
			prev = curr;
			curr = curr.next;
		}
		prev.next= null;
		
	}

	public void print(){
		Node curr = head;

		while(curr != null){
			System.out.print(curr.data + " ");
			curr = curr.next;
		}
	}
}
